src/System.Memory/ref/System.Memory.cs

```diff
@@ -12,21 +12,29 @@ public static partial class MemoryExtensions
         public static System.ReadOnlyMemory<char> AsMemory(this string text) { throw null; }
         public static System.ReadOnlyMemory<char> AsMemory(this string text, int start) { throw null; }
         public static System.ReadOnlyMemory<char> AsMemory(this string text, int start, int length) { throw null; }
+        public static System.ReadOnlyMemory<char> AsMemory(this string text, Index startIndex) { throw null; }
+        public static System.ReadOnlyMemory<char> AsMemory(this string text, Range range) { throw null; }
         public static System.Memory<T> AsMemory<T>(this System.ArraySegment<T> segment) { throw null; }
         public static System.Memory<T> AsMemory<T>(this System.ArraySegment<T> segment, int start) { throw null; }
         public static System.Memory<T> AsMemory<T>(this System.ArraySegment<T> segment, int start, int length) { throw null; }
         public static System.Memory<T> AsMemory<T>(this T[] array) { throw null; }
         public static System.Memory<T> AsMemory<T>(this T[] array, int start) { throw null; }
         public static System.Memory<T> AsMemory<T>(this T[] array, int start, int length) { throw null; }
+        public static System.Memory<T> AsMemory<T>(this T[] array, Index startIndex) { throw null; }
+        public static System.Memory<T> AsMemory<T>(this T[] array, Range range) { throw null; }
         public static System.ReadOnlySpan<char> AsSpan(this string text) { throw null; }
         public static System.ReadOnlySpan<char> AsSpan(this string text, int start) { throw null; }
         public static System.ReadOnlySpan<char> AsSpan(this string text, int start, int length) { throw null; }
         public static System.Span<T> AsSpan<T>(this System.ArraySegment<T> segment) { throw null; }
         public static System.Span<T> AsSpan<T>(this System.ArraySegment<T> segment, int start) { throw null; }
         public static System.Span<T> AsSpan<T>(this System.ArraySegment<T> segment, int start, int length) { throw null; }
+        public static Span<T> AsSpan<T>(this System.ArraySegment<T> segment, Index startIndex) { throw null; }
+        public static Span<T> AsSpan<T>(this System.ArraySegment<T> segment, Range range)  { throw null; }
         public static System.Span<T> AsSpan<T>(this T[] array) { throw null; }
         public static System.Span<T> AsSpan<T>(this T[] array, int start) { throw null; }
         public static System.Span<T> AsSpan<T>(this T[] array, int start, int length) { throw null; }
+        public static Span<T> AsSpan<T>(this T[] array, Index startIndex) { throw null; }
```

Comment:
If we are adding Index/Range overloads to AsSpan for array/segment, why not add it for string as well?

```C#
public static System.ReadOnlySpan<char> AsSpan(this string text, int start, int length) { throw null; }
```

Reply:
This is a good question, I don't think this was discussed and I don't see a strong reason why we shouldn't have it. but in general it is easy to achieve the same with the current APIs ```String.AsSpan().Slice(Index/Range)```

@terrajobst what you think about this one?